uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/BaseUnitTest.kt

```diff
@@ -0,0 +1,159 @@
+import org.isoron.uhabits.core.commands.CommandRunner
+import org.isoron.uhabits.core.database.Database
+import org.isoron.uhabits.core.database.DatabaseOpener
+import org.isoron.uhabits.core.database.JdbcDatabase
+import org.isoron.uhabits.core.database.MigrationHelper
+import org.isoron.uhabits.core.models.HabitList
+import org.isoron.uhabits.core.models.ModelFactory
+import org.isoron.uhabits.core.models.Timestamp
+import org.isoron.uhabits.core.models.memory.MemoryModelFactory
+import org.isoron.uhabits.core.tasks.SingleThreadTaskRunner
+import org.isoron.uhabits.core.test.HabitFixtures
+import org.isoron.uhabits.core.utils.DateUtils.Companion.getStartOfTodayCalendar
+import org.isoron.uhabits.core.utils.DateUtils.Companion.setFixedLocalTime
+import org.isoron.uhabits.core.utils.DateUtils.Companion.setStartDayOffset
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnitRunner
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.nio.file.Paths
+import java.sql.DriverManager
+import java.sql.SQLException
+
+@RunWith(MockitoJUnitRunner::class)
+open class BaseUnitTest {
+    protected open var habitList: HabitList? = null
+    protected var fixtures: HabitFixtures? = null
+    protected var modelFactory: ModelFactory? = null
+    protected var taskRunner: SingleThreadTaskRunner? = null
+    protected open var commandRunner: CommandRunner? = null
+    protected var databaseOpener: DatabaseOpener = object : DatabaseOpener {
+        override fun open(file: File): Database {
+            return try {
+                JdbcDatabase(
+                    DriverManager.getConnection(
+                        String.format(
+                            "jdbc:sqlite:%s",
+                            file.absolutePath
+                        )
+                    )
+                )
+            } catch (e: SQLException) {
+                throw RuntimeException(e)
+            }
+        }
+    }
+
+    @Before
+    @Throws(Exception::class)
+    open fun setUp() {
+        setFixedLocalTime(FIXED_LOCAL_TIME)
+        setStartDayOffset(0, 0)
+        val memoryModelFactory = MemoryModelFactory()
+        habitList = Mockito.spy(memoryModelFactory.buildHabitList())
+        fixtures = HabitFixtures(memoryModelFactory, habitList)
+        modelFactory = memoryModelFactory
+        taskRunner = SingleThreadTaskRunner()
+        commandRunner = CommandRunner(taskRunner!!)
+    }
+
+    @After
+    @Throws(Exception::class)
+    open fun tearDown() {
+        Mockito.validateMockitoUsage()
+        setFixedLocalTime(null)
+        setStartDayOffset(0, 0)
+    }
+
+    fun unixTime(year: Int, month: Int, day: Int): Long {
+        val cal = getStartOfTodayCalendar()
+        cal[year, month, day, 0, 0] = 0
```

Comment:
It seems like the automatic Kotlin conversion transforms `cal.set(year, month, day, hour, min, sec)` into `cal[year, month, day, hour, min] = sec`. Could you correct this?